controller/src/main.cpp

```diff
@@ -60,6 +60,7 @@ static ControllerStatus controller_status = ControllerStatus_init_zero;
 // Last-received status from the GUI.
 static GuiStatus gui_status = GuiStatus_init_zero;
 
+static Sensors sensors;
 static Controller controller;
```

Comment:
Do these need to be global variables?

Advantage of making them non-global is you don't need Sensors::Init() -- those init methods are always a trap (what happens if you  use the thing before initializing it?  In this case it's UB, you read the zeroes-array uninitialized.  This might even *work* most of the time.  You could fix it and add an assert, but that's kinda scary too!)

Reply:
Good point, made all the variables here non-static.